python/felis/check.py

```diff
@@ -67,6 +67,53 @@ def check_schema(self, schema_obj: _Mapping) -> None:
         _id = self._assert_id(schema_obj)
         self._check_visited(_id)
 
+    def check_schema_version(self, version_obj: Any, schema_obj: _Mapping) -> None:
+        """Validate contents of schema description object.
+
+        Parameters
+        ----------
+        version_obj : `Any`
+            Felis object (should be string or mapping) representing schema
+            version.
+        schema_obj : `Mapping` [ `str`, `Any` ]
+            Felis object (mapping) representing parent schema.
+
+        Raises
+        ------
+        TypeError
+            Raised if validation fails for expected types of items.
+        ValueError
+            Raised if validation fails for the content of the object.
+        """
+        if isinstance(version_obj, Mapping):
+            # "current" is required, other keys are optional.
+            possible_keys = {"current", "compatible", "read_compatible"}
+            if not possible_keys.issuperset(version_obj):
+                extra_keys = list(set(version_obj) - possible_keys)
+                logger.error(f"unexpected keys in schema version description: {extra_keys}")
+            if "current" not in version_obj:
+                raise ValueError(f"missing 'current' key in schema version description: {version_obj}")
+            if not isinstance(version_obj["current"], str):
+                raise TypeError(f"schema version 'current' value is not a string: {version_obj['current']!r}")
+            if (compatible := version_obj.get("compatible")) is not None:
+                if not isinstance(compatible, list):
+                    raise TypeError(f"schema version 'compatible' value is not a list: {compatible!r}")
+                for item in compatible:
+                    if not isinstance(item, str):
+                        raise TypeError(f"items in 'compatible' value are not strings: {compatible!r}")
+            if (read_compatible := version_obj.get("read_compatible")) is not None:
+                if not isinstance(read_compatible, list):
```

Comment:
All this code makes me dream of felis models being defined with Pydantic...

Reply:
I thought about Pydantic, but did not want to introduce the dependency just yet, there is a lot of code in this package that I still don't quite understand. We'll need another ticket to make use of Pydantic.